test/e2e/launch.e2e.test.js

```diff
@@ -1,108 +1,84 @@
-/* eslint-env mocha */
+const { _electron: electron } = require('playwright')
+const { test, expect } = require('@playwright/test')
 
-const { Application } = require('spectron')
-const electronPath = require('electron') // Require Electron = require(the binaries included in node_modules.)
 const path = require('path')
 const fs = require('fs-extra')
 const tmp = require('tmp')
-const delay = require('delay')
-const chai = require('chai')
-const dirtyChai = require('dirty-chai')
 const { makeRepository } = require('./utils/ipfsd')
 const portfinder = require('portfinder')
 
-const expect = chai.expect
-chai.use(dirtyChai)
-
-// To print the app logs, add the following to your test:
-//
-// const logs = await app.client.getMainProcessLogs()
-// logs.forEach(line => console.log(line))
-//
 async function getPort () {
   return portfinder.getPortPromise()
 }
 
-// Note: logs before "start daemon FINISHED" event are consumed
-// inside of `daemonReady`. To print them, pass DEBUG=true
-
-describe('Application launch', function () {
-  this.timeout(60000)
+test.describe.serial('Application launch', async () => {
   let app = null
 
-  afterEach(async function () {
-    if (app && app.isRunning()) {
-      await app.stop()
+  test.afterEach(async () => {
+    if (app) {
+      await app.close()
     }
   })
 
-  async function startApp ({
-    home = tmp.dirSync({ prefix: 'tmp_HOME_', unsafeCleanup: true }).name,
-    repoPath = path.join(home, '.ipfs')
-  }) {
-    app = new Application({
-      path: electronPath,
-      args: ['-r', path.join(__dirname, 'utils/include.js'), path.join(__dirname, '../../src/index.js')],
-      env: {
+  async function startApp ({ repoPath } = {}) {
+    const home = tmp.dirSync({ prefix: 'tmp_home_', unsafeCleanup: true }).name
+    if (!repoPath) {
+      repoPath = path.join(home, '.ipfs')
+    }
+    app = await electron.launch({
+      args: [path.join(__dirname, '../../src/index.js')],
+      env: Object.assign({}, process.env, {
         NODE_ENV: 'test',
         HOME: home,
         IPFS_PATH: repoPath
-      }
+      })
```

Comment:
:information_source: @hacdias this fixed the weird CI failure on Linux.

It was caused by the difference in how `env` field works here.
In  playwright,  passed env fully replaces `process.env` for greater control, so we were missing some variables created by X11/Xvfb.  This restores them, solving e2 on Linux.

Reply:
That makes sense. Thanks for figuring it out :)